modules/dag-l1/src/main/scala/org/tessellation/dag/l1/StateChannel.scala

```diff
@@ -67,7 +68,8 @@ class StateChannel[F[_]: Async: KryoSerializer: SecurityProvider: Random](
 
   private val ownRoundTriggerInput: Stream[F, OwnerBlockConsensusInput] = Stream
     .awakeEvery(5.seconds)
-    .evalFilter { _ =>
+    .evalTap(_ => semaphores.blockCreation.acquire)
```

Comment:
this semaphore is used here only. why can't you create it just before creating the StateChannel class, as we do with Ref's and similars?

Reply:
I can and initially I did, but then for some reason I thought that probably we will want to extract it somewhere, as we do with storages. I was wrong I guess ;)